goldens/foo/lib/foo.analyzer.json

```diff
@@ -11,6 +11,16 @@
                 "isField": true,
                 "isMethod": false,
                 "isStatic": false
+              },
+              "returnType": {
```

Comment:
This model for `properties` I think is going to fall apart here. We really don't want to be modeling all the possible properties for all the possible things with a single type?

For instance, `returnType` is really not what the type of a field is, there is a return type on the getter sure, but not the field (it just has a type).

When you are using this `Properties` class, you are also going to get terrible auto complete, with all possible things that might exist on all possible declaration kinds?